src/charm.py

```diff
@@ -165,14 +165,18 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             )
             event.defer()
             return
-        if self._certificates_relation_created:
+        if not self._csr_is_stored:
+            self._revoke_existing_certificates()
+            self._generate_csr()
+            self.unit.status = ActiveStatus()
+            return
+        elif self._certificates_relation_created:
             self._revoke_existing_certificates()
             self._generate_csr()
             self._request_certificate()
             self.unit.status = WaitingStatus("Waiting for certificate to be available")
         else:
-            self._generate_csr()
-            self.unit.status = ActiveStatus()
+            self.unit.status = WaitingStatus("Waiting for certificates relation to be created.")
```

Comment:
WaitingStatus should be for cases where we are waiting for something to be happen. BlockedStatus is used whenever we expect users to take action. In our case it seems like we're waiting for the user to take action.

Reply:
In the else clause, we are waiting for a relation to be created, so user need to relate it. So, it needs to be in BlockedStatus, right ? I will change it.